### ISSUE TITLE
[req] Manual search - add second Search box which removes results with entered text

### ISSUE THREAD

**Issue by ilike2burnthing** [Jackett/Jackett#7660]
_I had originally posted this as an edit to #6791 but I realise now it's completely different to what I had originally suggested._

Change the Dashboard Search box (not 'Query', but rather the secondary 'Search' which filters the results) to two boxes, with Search becoming Refine and the new box being Remove (or something to that effect).

Example taken from OP:
When searching for a show called Huge, not surprisingly you get a lot of porn, and filtering by '2010' just gives you a lot of 2010 porn instead. Being able to tell Jackett to remove results containing 'XXX', or anything else, would be a great help.

**Comment by garfield69** [comment]
or we could just implements a simple +string -string like most filters on web sites can do.

**Comment by ilike2burnthing** [comment]
Apologies for my ignorance, but would that essentially be the user typing, to use the earlier example, `-xxx` to remove all results with `xxx` in the name (and + would just be assumed in all other cases)?

If so, that would be a step in the direction of my OP. I just worried that on its own it might clash with users trying to refine by release group (e.g. `-EPSiLON` would remove results with `EPSiLON` in the name, so `=` and/or `~` would need to be implemented as well, or just `~!`). Or can you use `+-EPSiLON` to refine by `-EPSiLON`?

Sorry if this seems like I'm asking you to ELI5.

**Comment by cadatoiva** [comment]
You basically have the right idea there. Any term starting with - will be excluded like your suggestion above. Any that start with + (or don't start with -) must be in the result.

**Comment by ilike2burnthing** [comment]
Thanks for clearing that up.

So is `+-EPSiLON` possible, as a way to refine by `-EPSiLON`, rather than removing by `EPSiLON`? Or would would the aforementioned operators need to be implemented as well (e.g. `=-EPSiLON`)?

**Comment by cadatoiva** [comment]
When this gets implemented, I think it would be sufficient to just add +-EPSiLON. Basically the parser would check each word for a first character of +/- remove it, and add it to the appropriate list. If neither are found, then it acts as if it started with +

**Comment by ilike2burnthing** [comment]
Sounds good to me!

Wildcards can wait for another day lol

**Comment by ngosang** [comment]
This should be easy to do => https://datatables.net/examples/plug-ins/range_filtering.html

**Comment by cadatoiva** [comment]
I don't have much front-end experience, so I'll leave that to you @ngosang.

**Comment by wellloaded** [comment]
I really hope the negative filtering (-) to be looked into as it's a great feature to have!

**Comment by garfield69** [comment]
Jackett 0.14.447

**Comment by RoloSoze** [comment]
@garfield69 I'm assuming #8048 does not completely close this, correct? Wildcard and +string still need to be implemented?

**Comment by garfield69** [comment]
the plus is redundant as just adding a word in the search box is filtering in results with the word
and wildcards are not required as the search box does partials anyway.

**Comment by ilike2burnthing** [comment]
@RoloSoze +string is working, although the only case I've found so far where it's actually useful is searching for `-` i.e. type `+-` and only results containing `-` will appear (likewise `--` removes results containing `-`).

ngosang's comment (the one I linked to in the other thread), and my response to it, explains further about what can and can't be done.

**Comment by batuhankrmn** [comment]
Can this feature be modified as "or" logic instead of "and"? I mean when I want to filter the results for x265 codec., I can filter with +x265 but there is also HEVC tag which is commonly used for x265 codecs. So, it'd be really useful If we can filter the results like:

+x265 +HEVC: Gives the results contains HEVC or x265 in the torrent name.

For the current logic, If I filter with +x265 +HEVC It'll only give the results that contains both HEVC and x265 together. This will critically limit the results because if there is HEVC on title, there might not be x265, vice versa.

@garfield69 @ngosang @ilike2burnthing

**Comment by ilike2burnthing** [comment]
If we were to change it, we'd then end up with another request to change it back.

I often find myself using 5-10 filter words on large searches, and that would only work with AND logic.

You can always just filter by `x265` and then after by `HEVC` or `-x265 HEVC` to find any additional results.

**Comment by batuhankrmn** [comment]
I understand you completely. AND logic is mandatory for those scenarios. So, what about a new syntax like +(x265 hevc) ? The filter words between the ( ) will be filtered with OR logic. To clarification;

+(x265 hevc) = Gives the results contains HEVC or x265 in the torrent name.
-(x265 hevc) = Filters out the results contains HEVC or x265 in the torrent name.

I believe such a feature will bring a whole new functionality to Jackett.

**Comment by ilike2burnthing** [comment]
Feel free to open a new req for OR logic to be added like this.

It will be reliant on a contributor or maintainer having the time and ability to do so, so it won't be any time soon.

**Comment by batuhankrmn** [comment]
Exactly! I just tried Prowlarr and I can say that only plus feature is Prowlarr's custom filters. Jackett already has great filtering features that Prowlarr doesn't have but OR logic can make everything more perfect. I returned to Jackett after a week because I used to it, using the Jackett for years and recommending to everyone. If I can contribute in terms of coding, I definitely would but all I can do now is give feedback as an outside eye and make suggestions for improvement/enhancement.
